requirements/views.py

```diff
@@ -126,11 +126,9 @@ def progress(request, list_id, courses):
     in courses as a comma-separated list of subject IDs."""
     req = None
 
-    try:
```

Comment:
try/except is good here, because `json.loads` can throw an exception for badly-formatted input. Right about the other bug though.